src/sentry/tasks/post_process.py

```diff
@@ -595,6 +597,35 @@ def process_rules(job: PostProcessJob) -> None:
     return
 
 
+def process_code_mappings(job: PostProcessJob) -> None:
+    if job["is_reprocessed"]:
+        return
+
+    from sentry.tasks.derive_code_mappings import derive_code_mappings
+
+    try:
+        event = job["event"]
+        project = event.project
+
+        cache_key = f"code-mappings:{project.id}"
+        project_queued = cache.get(cache_key)
+        if project_queued is None:
+            cache.set(cache_key, True, 3600)
+            logger.info(
+                f"derive_code_mappings: Events from {project.id} will not have code mapping derivation until {timezone.now() + timedelta(hours=1)}"
+            )
+
+        if project_queued or not features.has(
```

Comment:
This feature check will incorporate the option / killswitch added in https://github.com/getsentry/getsentry/pull/8777. 

We don't need to add separate logic here, which is pretty neat!

Reply:
Should we check this earlier? or only check once an hour?
It's probably good here.

Reply:
Actually, this brings up a good point. We should add the killswitch check in `derive_code_mappings.py` as well. Even though it's a duplicate, we might want to stop all processing and right now, anything that's queued will still go through. I'll send a PR for that.